browser-switch/src/main/java/com/braintreepayments/browserswitch/BrowserSwitchFragment.java

```diff
@@ -158,8 +163,19 @@ private boolean isReturnUrlSetup() {
                 .addCategory(Intent.CATEGORY_DEFAULT)
                 .addCategory(Intent.CATEGORY_BROWSABLE);
 
-        List<ResolveInfo> activities = mContext.getPackageManager()
+        List<ResolveInfo> availableActivities = availableActivities(intent);
+
+        return availableActivities != null && availableActivities.size() == 1;
+    }
+
+    private boolean activityAvailable(Intent intent) {
+        List<ResolveInfo> activities = availableActivities(intent);
+
+        return activities != null && activities.size() > 0;
+    }
+
+    private List<ResolveInfo> availableActivities(Intent intent) {
+        return mContext.getPackageManager()
```

Comment:
We should always expect at least an empty list from this function. Sounds like `PackageManager#queryIntentActivities` does return an empty list.

>[If there are no matching activites, an empty list is returned](https://developer.android.com/reference/android/content/pm/PackageManager.html#queryIntentActivities(android.content.Intent,%20int))